GetStream/stream-video-android
# Improve video filters

### 🎯 Goal

- Add virtual background video filters
- Improve existing blur filters
- Refactor filter classes & add filter parameters
- Add separate artifact with filters & ML Kit dependency
- Add filter selection in Demo App
- Edit [docs](https://staging.getstream.io/video/docs/android/advanced/apply-video-filters/#adding-ai-filters) on video filters and [Installation](https://staging.getstream.io/video/docs/android/basics/installation/#video-filters) page.

### 🛠 Implementation details

- New module: `stream-video-android-filters-video`
- `@Keep` annotations for types in the new module
- Filter classes: `BlurredBackgroundVideoFilter` and `VirtualBackgroundVideoFilter` in new artifact
  - Filter tweaking parameters: `blurIntensity`, `backgroundImage`, `foregroundThreshold`
- New UI components (`StreamDrawableButton` and `StreamDrawableToggleButton`) and styles (`ButtonDrawableStyle` and `StreamButtonDrawableStyle`) in `stream-video-android-ui-compose`
- New _Video Filters_ horizontal menu in Demo App below _Reactions_ menu
- Added image options in Demo App as WebP images
- Added new content in docs and removed previous "blue background filter" section.

### 🎨 UI Changes

| Before | After |
| --- | --- |
| ![image](https://github.com/GetStream/stream-video-android/assets/65943217/f05c2559-1656-48af-b694-38c974f52d4d) | ![image](https://github.com/GetStream/stream-video-android/assets/65943217/fbf55438-162c-4540-b16c-00c55c978d49) |

### 🧪 Testing

- Run the demo app and use the blur and virtual backgrounds filters. Check the quality of the result.
- Check the docs [here](https://staging.getstream.io/video/docs/android/advanced/apply-video-filters/) and [here](https://staging.getstream.io/video/docs/android/basics/installation/#video-filters)
- Check code files
- Check new `stream-video-android-filters-video`, its configuration for Maven publish and correct marking of types with `@Keep`.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExd3U3cjlpNGNmOGF5dnVzYzN2ZWx4a256amFmYXI1NG9hcXQ5NXZtciZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/rDGgiLavSKLSw/giphy.gif)
